### DISCUSSION TITLE
libui-ng related versioning

### DISCUSSION THREAD

**Discussion by AndyObtiva** [kojix2/LibUI#57] [General]
Hi,

I just wanted to make a suggestion.

I was thinking that when LibUI starts supporting [libui-ng](https://github.com/libui-ng/libui-ng) new releases, it would do so through new minor versions (e.g. 0.1.0, 0.2.0, 0.3.0, etc...). That way, [glimmer-dsl-libui](https://github.com/AndyObtiva/glimmer-dsl-libui) would not break for users given that its LibUI gem dependency is fixed as `~> 0.0.15` (in case there were any breaking changes in [libui-ng](https://github.com/libui-ng/libui-ng) releases). That gives me the chance to test and upgrade [glimmer-dsl-libui](https://github.com/AndyObtiva/glimmer-dsl-libui) intentionally for each [libui-ng](https://github.com/libui-ng/libui-ng) release.

**Comment by kojix2** [comment]
That is a good idea.

**Comment by AndyObtiva** [comment]
I noticed you added rake task support for downloading any pre-release version of libui-ng, which is great.

I wonder if it would be a good idea for you to select an unreleased libui-ng version that you think is stable enough and make an alpha 0.1.x LibUI release with it (covering both x86 and arm64 on Mac) before RubyConf 2022 (Nov 29 - Dec 1, 2022). That way, before I present Glimmer DSL for LibUI at RubyConf 2022, I will make a Glimmer DSL for LibUI release that wraps your 0.1.x alpha release, and I will announce it at RubyConf 2022.

That would be good publicity for the projects!

**Comment by kojix2** [comment]
Yes, it would be great if we could do that. 
I am a little busy at the end of the year, and perhaps I will not make it. 
But let's give it a try.

**Comment by kojix2** [comment]
As you may have noticed, I have uploaded https://rubygems.org/gems/libui/versions/0.1.0.pre.0. I would expect it to work well.

Unfortunately, however, I should say that it was not ready in time for RubyConf. I am looking to improve #68 and add some platforms. Namely aarch64-linux (Raspberry Pi) and x86-mingw32.

**Comment by AndyObtiva** [comment]
Thank you for informing me. I just released glimmer-dsl-libui version 0.6.0.pre.0 that uses libui version 0.1.0.pre.0

`open_folder` now works!

I will announce this release at RubyConf 2022 and inform the people who wanted `open_folder` about it.

Cheers.

**Comment by AndyObtiva** [comment]
By the way, I tested and confirmed that all Glimmer DSL for LibUI examples work with LibUI version `0.1.0.pre.0`

As such, I think it is OK to release as `0.1.0` next.

LibUI in Ruby is alpha anyways (less than version `1.0`), so it is OK if it does not say `pre.x` after confirming that it works with all the C libui API calls in all examples.
